### ISSUE TITLE
Line chart's last value not selectable via marker

### ISSUE THREAD

**Issue by af-hku** [patrykandpatrick/vico#320]
**Describe the bug**
The last value of a line chart cannot be selected with the marker.

Conditions
* `HorizontalLayout.FullWidth` line chart
* Non-scrollable line chart (`isScrollEnabled = false`)
* Distance between points is really small (e.g. a lot of points, small chart or both)

**Module**
- [ ] `compose`
- [ ] `views`
- [ ] `compose-m2`
- [ ] `compose-m3`
- [x] `core`

**To reproduce**
* Download [vico v1.7.2 source code](https://github.com/patrykandpatrick/vico/releases/tag/v1.7.2)
* Make `com.patrykandpatrick.vico.sample.showcase.charts.Chart3` non-scrollable: 
```
ProvideChartStyle(rememberChartStyle(chartColors)) {
    Chart(
        ...
        chartScrollSpec = rememberChartScrollSpec(isScrollEnabled = false),
        ...
    )
}

```
* Run app and try to select last line chart value for `Chart3`

**Expected behavior**
It should be possible to select the last value using the marker.

**Screenshots**
![image](https://github.com/patrykandpatrick/vico/assets/71651622/ecc55a65-6709-4b35-925d-fc710e6f02b2)

**Android versions**
Android 10, Android 12

### Additional context
Problem lays within `com.patrykandpatrick.vico.core.chart.line.LineChart (334)`
* `if (x in bounds.left..bounds.right)` is false for the last value and the last value is therefore not added to `entryLocationMap` and cannot be selected via the marker
* e.g. Chart3: last value `x = 898.5938` BUT `bounds.right = 898.59375`

Cause:
`com.patrykandpatrick.vico.core.chart.line.LineChart (505)`
* `x = nextX ?: getDrawX(entry)` is executed and since it is the last value it goes into `getDrawX` which returns a value which is not inside the bounds.

**Comment by patrickmichalik** [comment]
Hello! Thanks for the bug report. Unfortunately, I’ve not been able to reproduce this—I’m able to get a marker to show up for the last entry. Could you share some information on your device’s display? Specifically, what is its resolution, and under “Developer options,” what is the value of “Smallest width”?

**Comment by af-hku** [comment]
You are right. I am not able to reproduce it using my emulators, either.
It must be some weird edge case. The problem is that this bug occurs on all of my devices with my real project. (300 data points, full width line chart)

Device, on which it can be reproduced with `Chart3`
* Samsung Galaxy S9
* 1440 x 2960 PX with (~ 570 pixel density)
* Smallest width: 384dp

**Comment by patrickmichalik** [comment]
I’ve managed to reproduce the problem on a Pixel 7. We’ll be looking into it.

**Comment by patrickmichalik** [comment]
Hello again! Could you take a look if this is fixed with Vico 1.9.2? It seems that the changes we’ve recently introduced have resolved the problem, even if this wasn’t entirely intentional. I’m now able to get a marker to show up for the last entry on the Pixel 7. Cheers!

**Comment by patrickmichalik** [comment]
Thanks for the update. I was able to reproduce the problem again, and [Vico 1.10.0](https://github.com/patrykandpatrick/vico/releases/tag/v1.10.0) resolves it. We’ve tested the fix rather extensively, so I’ll be closing this issue, but please feel free to let me know if everything looks good. In the unlikely case that the issue persists, we’ll reopen the issue. Cheers!
